src/cmd/migrate.ts

```diff
@@ -0,0 +1,106 @@
+/* eslint-disable no-await-in-loop */
+/* eslint-disable no-restricted-syntax */
+import { diff } from 'deep-diff'
+import { cloneDeep, get, isEqual, set, unset } from 'lodash'
+import { compare, valid } from 'semver'
+import { Argv } from 'yargs'
+import { $ } from 'zx'
+import { prepareEnvironment } from '../common/cli'
+import { OtomiDebugger, terminal } from '../common/debug'
+import { hfValues } from '../common/hf'
+import { getFilename, loadYaml, rootDir } from '../common/utils'
+import { writeValues } from '../common/values'
+import { BasicArguments, setParsedArgs } from '../common/yargs'
+import { askYesNo } from '../common/zx-enhance'
+
+const cmdName = getFilename(__filename)
+const debug: OtomiDebugger = terminal(cmdName)
+
+interface Change {
+  version: string
+  deletions?: Array<string>
+  locations?: Array<{
+    [oldLocation: string]: string
+  }>
+  mutations?: Array<{
+    [preMutation: string]: string
+  }>
+}
+
+export type Changes = Array<Change>
+
+export const moveGivenJsonPath = (yaml: Record<string, unknown> | undefined, lhs: string, rhs: string): void => {
+  const moveValue = get(yaml, lhs, 'err!')
+  if (yaml && unset(yaml, lhs)) set(yaml, rhs, moveValue)
+}
+
+export const inlineGucci = async (): Promise<string> => {
```

Comment:
I thought it was not relevant for this situation, since we only apply the inline template and nothing else. I looked at the shape of the input for function, and it is neither a path nor an object.